src/reconcile/rec_write.c

```diff
@@ -2125,15 +2150,28 @@ static int
 __rec_split_chunk_init(
     WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_CHUNK *chunk, size_t memsize)
 {
-	chunk->min_recno = WT_RECNO_OOB;
-	chunk->min_entries = 0;
-	/* Don't touch the key item memory, that memory is reused. */
-	chunk->min_key.size = 0;
-
 	chunk->recno = WT_RECNO_OOB;
-	chunk->entries = 0;
 	/* Don't touch the key item memory, that memory is reused. */
 	chunk->key.size = 0;
+	chunk->entries = 0;
+	chunk->oldest_start_ts = WT_TS_MAX;
+	chunk->newest_start_ts = chunk->newest_stop_ts = WT_TS_NONE;
+	if (!__wt_process.page_version_ts || r->page->type == WT_PAGE_COL_FIX) {
+		chunk->oldest_start_ts = WT_TS_NONE;
```

Comment:
The reason the initialization is reversed is that in the case of fixed-length column store or when building old-style pages, we don't want to accumulate an address timestamp triplet, we know from the start it's going to be "durable", and it's easier to set correct values initially than to correct the values later. When we are maintaining address timestamp triplets, then we want the values to be at the end of their respective ranges so they are corrected as we review key/value pairs.

I've abstracted out address timestamp initialization and added a comment in a88aa00, hopefully that clarifies things.

Also, I think the same change you flagged above (for `cell.i`), should be applied here as well. I've changed it so `newest_start` is initialized to `WT_TS_NONE` when we're **not** maintaining address timestamps.

Because there's no out-of-band timestamp value, it's not possible to use `WT_TS_NONE` as a check that timestamps aren't being used or aren't initialized. I'd flag any code you see doing that. (It makes debugging timestamps harder, but I don't think there's an alternative.)